### ISSUE TITLE
Fastify handler

### ISSUE THREAD

**Issue by jclaessens97** [sachinraja/trpc-playground#28]
I'm using tRPC with `fastify`, but I'm not able to find a way to use the fastify handler for the playground.

I tried mimicking the express example, but I noticed pretty quickly that the adapter for fastify works in a different way (`getFastifyPlugin`?). I tried importing that but whatever I try to import using `import { getFastifyPlugin } from 'trpc-playground/handlers/fastify';` it's not working.

Inside `fastify.d.ts` however I can see the following:

```typescript
import { TrpcPlaygroundConfig } from '@trpc-playground/types';
import { FastifyPluginCallback } from 'fastify';
export declare const getFastifyPlugin: (config: TrpcPlaygroundConfig) => Promise<FastifyPluginCallback>;
```

Any hints on what I'm doing wrong?

**Comment by jclaessens97** [comment]
I just noted [PR#27](https://github.com/sachinraja/trpc-playground/pull/27), and it seems that the fix is not published yet on npm?

**Comment by jclaessens97** [comment]
If I do the same fix in the package, usage is still unclear to me.

I used it as follows:

```typescript
import fp from 'fastify-plugin';
import { fastifyTRPCPlugin } from '@trpc/server/adapters/fastify';
import { getFastifyPlugin } from 'trpc-playground/handlers/fastify';
import { appRouter } from '../router';
import { createContext } from '../context';

const trpcEndpoint = '/trpc';

export default fp(async (fastify) => {
  await fastify.register(fastifyTRPCPlugin, {
    prefix: trpcEndpoint,
    trpcOptions: {
      router: appRouter,
      context: createContext,
    },
  });

  const playgroundPlugin = await getFastifyPlugin({
    trpcApiEndpoint: trpcEndpoint,
    playgroundEndpoint: '/playground',
    router: appRouter,
  });
  await fastify.register(playgroundPlugin);
});
```

But it doesn't create an endpoint for the playground.

**Comment by sachinraja** [comment]
Thanks for reporting, will investigate when I get some time. It might just be broken.

**Comment by codyfyi** [comment]
@jclaessens97  - I've been using it without issue. I'm on fastify `^4.5.3`. Here is my script:

`
import fastify from 'fastify'
import {fastifyTRPCPlugin} from '@trpc/server/adapters/fastify'
import {getFastifyPlugin} from 'trpc-playground/handlers/fastify'
import {router} from '../routers/_router'

const TRPC_PORT = 4000
const TRPC_ENDPOINT = '/rpc'
const TRPC_PLAYGROUND_ENDPOINT = '/playground'

async function run() {
  const server = fastify()

  await server
    .register(fastifyTRPCPlugin, {prefix: TRPC_ENDPOINT, trpcOptions: {router: router}})
    .register(
      getFastifyPlugin({
        trpcApiEndpoint: TRPC_ENDPOINT,
        playgroundEndpoint: TRPC_PLAYGROUND_ENDPOINT,
        router: router
      }),
      {prefix: TRPC_PLAYGROUND_ENDPOINT}
    )

  try {
    await server.listen({port: 4000})
    console.log(`tRPC server ready at http://localhost:${TRPC_PORT}${TRPC_ENDPOINT}`)
    console.log(`tRPC playground ready at http://localhost:${TRPC_PORT}${TRPC_PLAYGROUND_ENDPOINT}`)
  } catch (err) {
    server.log.error(err)
    process.exit(1)
  }
}

run()
`

I'm using Brave/Chrome. I did notice Safari just loads a white screen and I get this error in the console:
`Cross-origin redirection to https://cdn.jsdelivr.net/npm/@trpc-playground/html@1.0.0-next.0/dist/assets/index.70ae6946.js denied by Cross-Origin Resource Sharing policy: Origin http://localhost:4000 is not allowed by Access-Control-Allow-Origin. Status code: 302
http://cdn.jsdelivr.net/npm/@trpc-playground/html@1.0.0-next.0/dist/assets/index.70ae6946.js`

I suspect you can fix Safari by adding a `@fastify/cors` configuration, but I haven't played with it.

**Comment by MatsDK** [comment]
@jclaessens97 I was able the resolve this issue by instead of writing:
```typescript
await fastify.register(playgroundPlugin);
```
writing this:
```typescript
await fastify.register(playgroundPlugin, { prefix: '/playground' });
```
Hopefully this helps.

**Comment by sachinraja** [comment]
published #27 in `0.4.4`
